inffast.c

```diff
@@ -107,10 +108,11 @@ void ZLIB_INTERNAL inflate_fast(z_stream *strm, unsigned long start) {
        input data or output space */
     do {
         if (bits < 15) {
-            hold += (uint32_t)(*in++) << bits;
-            bits += 8;
-            hold += (uint32_t)(*in++) << bits;
-            bits += 8;
+            unsigned short chunk;
+            __builtin_memcpy(&chunk, in, sizeof(unsigned short));
+            hold += (chunk << bits);
```

Comment:
It might be worth putting a generic `uint16_t` getter in a header file so it can be implemented differently depending on endianness and unaligned load support, but I couldn't demonstrate any real performance gain over handful of files with this technique anyway.  Though it was the first change I tried, and it's buried under a lot of other performance problems, probably including branch misprediction.